### PR TITLE
Fix user access roles from displaying business

### PR DESCRIPTION
**Pull request by ConorWebb96**
## Description
Changed business label to premium whenever the user doesn't have the correct permissions.

## Screenshots
Screenshot 2024-08-13 at 11 29 05

## Launchcontrol
Minor fix to UI. Access was showing business whenever user didn't have a high subscription tier. Should be showing premium.

### CODE REVIEW

**Comment by ConorWebb96** [comment]
Saw a thread as well as checked the pricing page and this is actually an enterprise tier feature. Updated UI, added some additional if statements to reflect this.

## Premium
Screenshot 2024-08-13 at 12 20 10

## UI
Screenshot 2024-08-13 at 12 20 54

## Enterprise
Screenshot 2024-08-13 at 12 38 46

## UI
Screenshot 2024-08-13 at 12 38 33

File: packages/builder/src/components/common/RoleSelect.svelte [hunk 1]
```diff
@@ -24,6 +24,7 @@
 
   const dispatch = createEventDispatcher()
   const RemoveID = "remove"
+  const subType = $licensing.license.plan.type
```
Comment (melohagan) [inline packages/builder/src/components/common/RoleSelect.svelte:27]: NAB: Just double check there's no null-pointer risk here

File: packages/builder/src/components/common/RoleSelect.svelte [hunk 2]
```diff
@@ -68,13 +69,19 @@
     }))
 
     // Add creator if required
-    if (allowCreator) {
+    if (
+      allowCreator ||
+      subType === Constants.PlanType.ENTERPRISE ||
+      subType === Constants.PlanType.ENTERPRISE_BASIC
```
Comment (melohagan) [inline packages/builder/src/components/common/RoleSelect.svelte:75]: I think you need to account for `ENTERPRISE_BASIC_TRIAL` as well

File: packages/builder/src/components/common/RoleSelect.svelte [hunk 3]
```diff
@@ -68,13 +69,19 @@
...
+    ) {
       options.unshift({
         _id: Constants.Roles.CREATOR,
         name: "Can edit",
         tag:
-          !$licensing.perAppBuildersEnabled &&
-          capitalise(Constants.PlanType.BUSINESS),
+          subType === Constants.PlanType.ENTERPRISE ||
+          subType === Constants.PlanType.ENTERPRISE_BASIC
+            ? null
```
Comment (melohagan) [inline packages/builder/src/components/common/RoleSelect.svelte:83]: Same as above `ENTERPRISE_BASIC_TRIAL`

File: packages/builder/src/components/common/RoleSelect.svelte [hunk 4]
```diff
@@ -134,9 +141,16 @@
...
     getOptionColour={getColor}
     getOptionIcon={getIcon}
-    isOptionEnabled={option =>
-      option._id !== Constants.Roles.CREATOR ||
-      $licensing.perAppBuildersEnabled}
+    isOptionEnabled={option => {
+      if (option._id === Constants.Roles.CREATOR) {
+        return (
+          subType === Constants.PlanType.ENTERPRISE ||
+          (subType === Constants.PlanType.ENTERPRISE_BASIC_TRIAL &&
```
Comment (melohagan) [inline packages/builder/src/components/common/RoleSelect.svelte:148]: This time you need `ENTERPRISE_BASIC` as well? And why is ENTERPRISE_BASIC_TRIAL and `perAppBuildersEnabled` grouped together here?

File: packages/builder/src/components/common/RoleSelect.svelte [hunk 5]
```diff
@@ -154,10 +168,16 @@
...
     getOptionColour={getColor}
     getOptionIcon={getIcon}
-    isOptionEnabled={option =>
-      (option._id !== Constants.Roles.CREATOR ||
-        $licensing.perAppBuildersEnabled) &&
-      option.enabled !== false}
+    isOptionEnabled={option => {
+      if (option._id === Constants.Roles.CREATOR) {
+        return (
+          subType === Constants.PlanType.ENTERPRISE ||
```
Comment (melohagan) [inline packages/builder/src/components/common/RoleSelect.svelte:174]: Same as above - could this duplicate code be moved into a shared function?

**Review by melohagan — CHANGES_REQUESTED** [review]

**Review by shogunpurple — CHANGES_REQUESTED** [review]
Just need to update enterprise basic trial key and it should be good to merge

File: packages/builder/src/components/common/RoleSelect.svelte [hunk 6]
```diff
@@ -117,6 +118,14 @@
       dispatch("change", e.detail)
     }
   }
+
+  function isEnterprisePlan(subType) {
+    return (
+      subType === Constants.PlanType.ENTERPRISE ||
+      subType === Constants.PlanType.ENTERPRISE_BASIC ||
+      subType === Constants.PlanType.ENTERPRISE_BASIC_trial
```
[the remark this answers is not in the document]
Comment (ConorWebb96) [reply packages/builder/src/components/common/RoleSelect.svelte]: Cheers and fixed 👍

**Review by shogunpurple — APPROVED** [review]
LGTM

**Pull request merged**